be/src/runtime/fragment_mgr.cpp

```diff
@@ -877,32 +887,23 @@ Status FragmentMgr::exec_plan_fragment(const TPipelineFragmentParams& params,
             }
             query_ctx->fragment_ids.push_back(fragment_instance_id);
         }
+
         START_AND_SCOPE_SPAN(tracer, span, "exec_instance");
         span->SetAttribute("instance_id", print_id(fragment_instance_id));
 
-        std::shared_ptr<FragmentExecState> exec_state(new FragmentExecState(
-                query_ctx->query_id, fragment_instance_id, local_params.backend_num, _exec_env,
-                query_ctx,
-                std::bind<void>(std::mem_fn(&FragmentMgr::coordinator_callback), this,
-                                std::placeholders::_1)));
-        if (params.__isset.need_wait_execution_trigger && params.need_wait_execution_trigger) {
-            // set need_wait_execution_trigger means this instance will not actually being executed
-            // until the execPlanFragmentStart RPC trigger to start it.
-            exec_state->set_need_wait_execution_trigger();
+        if (!(params.__isset.need_wait_execution_trigger && params.need_wait_execution_trigger)) {
+            query_ctx->set_ready_to_execute_only();
```

Comment:
it is wrong.
exec state is instance level. query ctx is query level, it contains many instances.